### PR TITLE
chore: bump vyper version

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

Upgrading the vault version to the latest version of vyper (0.3.0)

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/Vault.vy [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# @version 0.2.16
+# @version 0.3.0
```
Comment (fubuloubu) [inline contracts/Vault.vy:1]: v0.3.1 should be released today, should probably try that out

Comment (pandadefi) [reply contracts/Vault.vy]: IncompatibleVyperVersion: No installable vyper version matching '0.3.1' in 'contracts/Vault.vy'

**Comment by fubuloubu** [comment]
Should work now

**Comment by pandadefi** [comment]
> Should work now

still not working

**Comment by fubuloubu** [comment]
hmm, works for me... is your brownie up to date?

I see it reduced Vault.vy from 20,397B to 17,951B, a 12% reduction!

**Comment by pandadefi** [comment]
> hmm, works for me... is your brownie up to date?
> 
> I see it reduced Vault.vy from 20,397B to 17,951B, a 12% reduction!

Test are ok, let's merge this one

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
> > hmm, works for me... is your brownie up to date?
> > I see it reduced Vault.vy from 20,397B to 17,951B, a 12% reduction!
> 
> Test are ok, let's merge this one

Please test this with apetax first

**Pull request merged**